packages/arb-token-bridge-ui/src/components/TransferPanel/AdvancedSettings.tsx

```diff
@@ -1,11 +1,63 @@
 import { useEffect, useState } from 'react'
+import { isAddress } from 'ethers/lib/utils'
+import { Provider } from '@ethersproject/providers'
 import { ChevronDownIcon, ChevronUpIcon } from '@heroicons/react/24/outline'
 
 import { useAppState } from '../../state'
 import { useAccountType } from '../../hooks/useAccountType'
+import { addressIsSmartContract } from '../../util/AddressUtils'
+import { useNetworksAndSigners } from '../../hooks/useNetworksAndSigners'
 
-export enum AdvancedSettingsErrors {
-  INVALID_ADDRESS = 'The destination address is not valid.'
+export enum DestinationAddressErrors {
+  INVALID_ADDRESS = 'The destination address is not a valid address.',
+  REQUIRED_ADDRESS = 'The destination address is required.'
+}
+
+enum DestinationAddressWarnings {
+  CONTRACT_ADDRESS = 'The destination address is a contract address. Please make sure it is a valid wallet address.'
```

Comment:
maybe this change, because it's definitely a valid address as we were able to check it's a contract

```suggestion
  CONTRACT_ADDRESS = 'The destination address is a contract address. Please make sure it is the right address.'
```